### ISSUE TITLE
Do we really need to use chromedriver and selenium for searching?

### ISSUE THREAD

**Issue by ankitkalia1195** [rafayet-monon/google-search-extractor#25]
What are the advantages of using the chromedriver, over using simpler implementations like Net::HTTP? IMO the chromedriver/selenium is pretty heavy implentation for this excercise, and they should be used for automated testing only.

**Comment by rafayet-monon** [comment]
Hi,

Thanks for raising the issue. Yes chromedriver/selenium is pretty heavy and my first try was using `Net::HTTP` and after that I also tried `Nokogiri`. But the problem I faced while using them was those tools cannot work with javascript. 
One of the requirement of the project in `3(c)` was to get _Total of search results for this keywords e.g. About ​21,600,000​ results (0.42 seconds)_ . While using `Net::HTTP` or `Nokogiri` I wasn't able to access that particular portion from google search page. My guess was it was using javascript to show this portion of the page. Then I tried with chromedriver and selenium so that I can get the complete page executed in browser which has javascript and get the data. That is the reason for using chromedriver/selenium. I am open to suggestion and will replace selenium if something less heavy tool can be used to accomplish this.
